### ISSUE TITLE
Error when trying to connect to .Net Core API secured with Azure AD using AadHttpClient from SharePoint app

### ISSUE THREAD

**Issue by BG-PR** [SharePoint/sp-dev-docs#3646]
- [x] Question
_SPFx 
SharePoint Online / Office 365_

Hello,
I have a custom .Net core API uploaded to Azure. I need to connect the secure Azure AD API with the SharePoint client-side web parts (using SharePoint framework v 1.7). To connect to Azure AD I use AadHttpClient.

I receive the following error when opening API management page:
**[HTTP]:500 - [CorrelationId] {} [Version]:16.0.0.8713 - Unknown Error**
I see in the development tool that the error comes from: Request URL: tenant-admin.sharepoint.com/_api/SPOInternalUseOnly.SPOWebAppServicePrincipal/PermissionGrants
I'm a global administrator.

When I try to approve the permission request it shows an error:
**[HTTP]:400 - [CorrelationId] {}: [Version]:16.0.0.8713 - The service principal for permssion request {projectName}-client-side-solution could not be found.**
I try to approve it via Azure powershell Approve-SPOTenantServicePrincipalPermissionRequest but I receive the same error _(The service principle for permssion request ....)_

package-solution.json contains:
```
    "webApiPermissionRequests": [
      {
        "resource": "{displayName of the registered app}",
        "scope": "user_impersonation"
      } ]
```

In Azure app service settings -> Authentication/Authorizaion ->  Active Directory Authentication --> App Service Authentication is On -> Express -> Use Existing App .
When I execute I see my service registered there.
`az ad sp list --query "sort_by([*].{displayName: displayName, objectId: objectId}, &displayName)" -o table`
I also try to execute Enable-SPOTenantServicePrincipal but receive an error again : Unknown Error

_I use free trial subscription in Azure_
 Do you have any ideas?

**Comment by patmill** [comment]
Just to double check - you are using the native SharePoint mobile app?  That is currently not supported, but we are working on getting the supported as soon as possible.

**Comment by BG-PR** [comment]
It is not a mobile app. I use SPFx React application and deploy the package to SharePoint App Catalog in  SharePoint online.

**Comment by BG-PR** [comment]
Is there a way to check if "SharePoint Online Client Extensibility" principle service is available in aad.portal.azure.com? 
I executed executed Enable-SPOTenantServicePrincipal but received an error: Unknown Error
When I go to "Azure Active Directory" --> App registrations --> All apps I don't see it.
When I go to "Azure Active Directory" --> Enterprise applications --> I can see it but don't know if it is enabled.

**Comment by andrewconnell** [comment]
@BG-PR said:

> `"resource": "{displayName of the registered app}"`

Can you provide the value you have listed here in your code? This should be the name of the Azure AD app you are *calling*... not your app. For instance, **Microsoft Graph**

@BG-PR said:

> Is there a way to check if "SharePoint Online Client Extensibility" principle service is available in aad.portal.azure.com?

Yup, for instance, if you were calling the Microsoft Graph
- https://aad.azure.portal.com
- ** Azure Active Directory > Enterprise Applications**
- Change the filter to **All Applications**
- Select **GraphAggregatorService**
- Select **Properties** in the menu
- Notice **Name** = **Microsoft Graph**
  - This is what you use in the `resource` property

**Comment by BG-PR** [comment]
Hello,
I fixed my problem. It turns out that the "SharePoint Online Client Extensibility Web Application Principal" was in 'deleted' list. I used azure powershell to check and restore:
```
Connect-AzureAD
Get-AzureADDeletedApplication
Restore-AzureADDeletedApplication -ObjectId <your_object_id>
Enable-SPOTenantServicePrincipal --> no error anymore
```
See: https://github.com/SharePoint/sp-dev-docs/issues/2522
I 
In App registration -> API permissions I had to click ''Granting admin consent for ***". 
After restoring the mentioned principle was visible in App registrations. I was able to approve the permission request. No errors. 
The "resource" value was correct - the name of my uploaded service.
